sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -547,6 +547,10 @@ valueExpression
     | left=valueExpression comparisonOperator right=valueExpression                          #comparison
     ;
 
+builtinOpExpression
+    : primaryExpression (CONCAT_PIPE primaryExpression)*                                       #concat
```

Comment:
Should we move this to the head of the primaryExpression rule? That seems easier.

I am also trying to figure how this works with other binary operators, for example: `a + b || c`.